src/compiler/types.ts

```diff
@@ -4664,10 +4664,11 @@ namespace ts {
         UseAliasDefinedOutsideCurrentScope      = 1 << 14,  // Allow non-visible aliases
         UseSingleQuotesForStringLiteralType     = 1 << 28,  // Use single quotes for string literal type
         NoTypeReduction                         = 1 << 29,  // Don't call getReducedType
+        OmitThisParameter                       = 1 << 30,

```

Comment:
@weswigham @rbuckton could one of you take a look at this and see if we're ok with adding this flag? I'm also not sure we can use `1 << 30`. `1 << 25` seems to be free though, but again, I'm not sure if we're ok with adding a flag here.